src/averageAges.js

```diff
@@ -55,7 +75,22 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(person => people.some(child => child.name === person.mother)
+    && person.sex === 'm')
+    : people.filter(person =>
+      people.some(child => child.name === person.mother));
+
+  const difference = children.map(child => {
+    const motherOfChild = people.find(mother => mother.name === child.mother);
+
+    return child.born - motherOfChild.born;
+  });
+
+  const ages = difference.reduce((total, age) => total + age);
+  const differenceAges = ages / difference.length;
```

Comment:
The same remark - just return divide operation and don't create redundant variable. Or you can return ages - just add dividing on  'difference.length' after reduce: difference.reduce((total, age) => total + age) / difference.length;